components/mediation-initializer/org.wso2.carbon.mediation.transport.handlers/src/main/java/org/wso2/carbon/mediation/transport/handlers/requestprocessors/swagger/format/CarbonServerConfig.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mediation.transport.handlers.requestprocessors.swagger.format;
+
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.description.TransportInDescription;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.rest.API;
+import org.apache.synapse.rest.RESTConstants;
+import org.wso2.carbon.mediation.commons.rest.api.swagger.ServerConfig;
+import org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants;
+import org.wso2.carbon.utils.NetworkUtils;
+
+import java.net.SocketException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class CarbonServerConfig implements ServerConfig {
```

Comment:
Make singleton by getting the axisconfig from the data holder if possible

Reply:
Since there is already common singleton holding Axis config, we don't need to make this singleton we could use that within CarbonServerConfig. Did improvement in latest commit